### ISSUE TITLE
The algorithm to distribute memory is wrong 

### ISSUE THREAD

**Issue by MatiasVara** [MatiasVara/torokernel#183]
The current algorithm assign RESERVED regions.

**Comment by MatiasVara** [comment]
The limit address is also wrong.

**Comment by MatiasVara** [comment]
```
[23/03/2018-15:36:23] CPU0 Thread#0 ProcessInit: LocalCpuSpeed: 3577 Mhz, Cores: 1
[23/03/2018-15:36:23] CPU0 Thread#0 InitializeDirectory Chunk: 0x800000 Size: 527302656
[23/03/2018-15:36:23] CPU0 Thread#0 InitializeDirectory first initialization
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Chunk: 0x83C0F0 Size: 527056656 bytes
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x83C0F8, List: 0x414D78, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x83C0F8 SizeSX: 268435456 Remaining ChunkSize: 258621192
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1083C100, List: 0x414D58, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1083C100 SizeSX: 134217728 Remaining ChunkSize: 124403456
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1883C108, List: 0x414D38, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1883C108 SizeSX: 67108864 Remaining ChunkSize: 57294584
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1C83C110, List: 0x414D18, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1C83C110 SizeSX: 33554432 Remaining ChunkSize: 23740144
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1E83C118, List: 0x414CF8, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1E83C118 SizeSX: 16777216 Remaining ChunkSize: 6962920
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1F83C120, List: 0x414CB8, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1F83C120 SizeSX: 4194304 Remaining ChunkSize: 2768608
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FC3C128, List: 0x414C98, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FC3C128 SizeSX: 2097152 Remaining ChunkSize: 671448
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FE3C130, List: 0x414C58, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FE3C130 SizeSX: 524288 Remaining ChunkSize: 147152
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FEBC138, List: 0x414C18, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FEBC138 SizeSX: 131072 Remaining ChunkSize: 16072
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FEDC140, List: 0x414B98, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FEDC140 SizeSX: 8192 Remaining ChunkSize: 7872
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FEDE148, List: 0x414B78, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FEDE148 SizeSX: 4096 Remaining ChunkSize: 3768
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FEDF150, List: 0x414B58, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FEDF150 SizeSX: 2048 Remaining ChunkSize: 1712
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FEDF958, List: 0x414B38, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FEDF958 SizeSX: 1024 Remaining ChunkSize: 680
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FEDFD60, List: 0x414B18, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FEDFD60 SizeSX: 512 Remaining ChunkSize: 160
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FEDFF68, List: 0x414AD8, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FEDFF68 SizeSX: 128 Remaining ChunkSize: 24
[23/03/2018-15:36:23] CPU0 Thread#0 BlockListAdd: Chunk: 0x1FEDFFF0, List: 0x414A78, Count: 1
[23/03/2018-15:36:23] CPU0 Thread#0 SplitChunk Add Block (marked as free): 0x1FEDFFF0 SizeSX: 16 Remaining ChunkSize: 0

```

**Comment by MatiasVara** [comment]
Amount and Counter are completely wrong.

**Comment by MatiasVara** [comment]
The available memory is all the physical memory after the first MB.

**Comment by MatiasVara** [comment]
To create a ticket to investigate when more than 8Gb is assigned.

**Comment by MatiasVara** [comment]
I replaced the whole algorithm with a simpler one that only uses the first block of memory after the $100000 address. This should be fixed for a machine with more than 3.5Gb

**Comment by MatiasVara** [comment]
```
2 Gb
[24/03/2018-11:11:49] CPU0 Thread#0 Memory Map
[24/03/2018-11:11:49] CPU0 Thread#0 0x:0x9FBFF, Flags: Free
[24/03/2018-11:11:49] CPU0 Thread#0 0x9FC00:0x9FFFF, Flags: Reserved
[24/03/2018-11:11:49] CPU0 Thread#0 0xF0000:0xFFFFF, Flags: Reserved
[24/03/2018-11:11:49] CPU0 Thread#0 0x100000:0x7FFDFFFF, Flags: Free
[24/03/2018-11:11:49] CPU0 Thread#0 0x7FFE0000:0x7FFFFFFF, Flags: Reserved
[24/03/2018-11:11:49] CPU0 Thread#0 0xFFFC0000:0xFFFFFFFF, Flags: Reserved
```
